src/DotNet.Testcontainers/Containers/Modules/TestcontainersContainer.cs

```diff
@@ -93,7 +93,7 @@ public string Hostname
               return "localhost";
             }
           default:
-            return this.IpAddress;
+            return this.configuration.Hostname;
```

Comment:
I guess you mean this case only. I'm not so sure about other cases.

Reply:
Yep, but I don't think it will resolve the name immediately. At least in my small example it didn't.